packages/next-codemod/bin/transform.ts

```diff
@@ -126,36 +126,57 @@ export async function runTransform(
 
   console.log(`Executing command: jscodeshift ${args.join(' ')}`)
 
-  const result = execa.sync(jscodeshiftExecutable, args, {
-    stdio: 'inherit',
-    stripFinalNewline: false,
+  const execaChildProcess = execa(jscodeshiftExecutable, args, {
+    // include ANSI color codes
+    env: { FORCE_COLOR: 'true' },
```

Comment:
Is `execa` overriding `process.env` like the built-in `childProcess.exec` or merging?

In terminals without ANSI support, we'd produce unreadable output.

Our detection should work for ANSI and no ANSI support.

Reply:
> Is execa overriding process.env like the built-in childProcess.exec or merging?

Extends (merges), unless `extendEnv` is set to false.

https://github.com/sindresorhus/execa/blob/a4d13df33072740a3620ba35491199a5f9abda03/docs/api.md?plain=1#L846

Reply:
`FORCE_COLOR` is handled at Node.js side: https://github.com/nodejs/node/blob/7ae193d19f15a92578a7e7cc088e320a65788df0/lib/internal/util/colors.js#L19

`execa` just passes the `env` to Node.js `spawn`

Reply:
> In terminals without ANSI support, we'd produce unreadable output.

Q: You mean Node.js `spawn` will produce unreadable in the same condition?

FYI `FORCE_COLOR` is handled at Node.js side: https://github.com/nodejs/node/blob/7ae193d19f15a92578a7e7cc088e320a65788df0/lib/internal/util/colors.js#L19

`execa` just passes the `env` to Node.js `spawn`.

Reply:
We should set `FORCE_COLOR` conditionally if `process.stdout.isTTY`

Reply:
> Is that based on this?

What is based on this? I suggest reading Node.js docs instead of diving into source code to understand how things work. Especially on non-trivial codebases like Node.js some internal code may not be the correct thing to look at.